server/neptune/gateway/event/source/project_finder.go

```diff
@@ -0,0 +1,271 @@
+type DefaultProjectFinder struct{}
+
+// See ProjectFinder.DetermineProjects.
+func (p *DefaultProjectFinder) DetermineProjects(modifiedFiles []string, repoFullName string, absRepoDir string, autoplanFileList string) []models.Project {
+	var projects []models.Project
+
+	modifiedTerraformFiles := p.filterToFileList(modifiedFiles, autoplanFileList)
+	if len(modifiedTerraformFiles) == 0 {
+		return projects
+	}
+
+	var dirs []string
+	for _, modifiedFile := range modifiedTerraformFiles {
+		projectDir := p.getProjectDir(modifiedFile, absRepoDir)
+		if projectDir != "" {
+			dirs = append(dirs, projectDir)
+		}
+	}
+	uniqueDirs := p.unique(dirs)
+
+	// The list of modified files will include files that were deleted. We still
+	// want to run plan if a file was deleted since that often results in a
+	// change however we want to remove directories that have been completely
+	// deleted.
+	exists := p.removeNonExistingDirs(uniqueDirs, absRepoDir)
+
+	for _, p := range exists {
+		projects = append(projects, models.NewProject(repoFullName, p))
+	}
+	return projects
+}
+
+// See ProjectFinder.DetermineProjectsViaConfig.
+func (p *DefaultProjectFinder) DetermineProjectsViaConfig(modifiedFiles []string, config valid.RepoCfg, absRepoDir string) ([]valid.Project, error) {
+	var projects []valid.Project
+	for _, project := range config.Projects {
+		var whenModifiedRelToRepoRoot []string
+		for _, wm := range project.Autoplan.WhenModified {
+			wm = strings.TrimSpace(wm)
+			// An exclusion uses a '!' at the beginning. If it's there, we need
+			// to remove it, then add in the project path, then add it back.
+			exclusion := false
+			if wm != "" && wm[0] == '!' {
+				wm = wm[1:]
+				exclusion = true
+			}
+
+			// Prepend project dir to when modified patterns because the patterns
+			// are relative to the project dirs but our list of modified files is
+			// relative to the repo root.
+			wmRelPath := filepath.Join(project.Dir, wm)
+			if exclusion {
+				wmRelPath = "!" + wmRelPath
+			}
+			whenModifiedRelToRepoRoot = append(whenModifiedRelToRepoRoot, wmRelPath)
+		}
+		pm, err := fileutils.NewPatternMatcher(whenModifiedRelToRepoRoot)
+		if err != nil {
+			return nil, errors.Wrapf(err, "matching modified files with patterns: %v", project.Autoplan.WhenModified)
+		}
+
+		// If any of the modified files matches the pattern then this project is
+		// considered modified.
+		for _, file := range modifiedFiles {
+			match, err := pm.Matches(file)
+			if err != nil {
+				continue
+			}
+			if match {
+				// If we're checking using an atlantis.yaml file we downloaded
+				// directly from the repo (when doing a no-clone check) then
+				// absRepoDir will be empty. Since we didn't clone the repo
+				// yet we can't do this check. If there was a file modified
+				// in a deleted directory then when we finally do clone the repo
+				// we'll call this function again and then we'll detect the
+				// directory was deleted.
```

Comment:
I don't think this is applicable anymore, we should be able to remove the check.